### PR TITLE
Improve Gradle build scripts

### PR DESCRIPTION
**Pull request by DRSchlaubi**
- Use BOMs to avoid duplicated versions
- Use Gradle application plugins over shadowJar
- Use precompiled buildscripts
- Only apply common dependencies to common project
- Make gradlew executable﻿

### CODE REVIEW

File: docker/run.sh [hunk 1]
```diff
@@ -41,4 +41,4 @@ if [[ -z "${WINTERFOX_DEDI_NODE:-}" ]]
 
 JVM_ARGS+=("$@")
 
-java -jar /app/noelware/nino/Nino.jar $JVM_ARGS
+java -jar /app/noelware/nino/bot/bin/bot $JVM_ARGS
```
Comment (NyCodeGHG) [inline docker/run.sh:44]:
```suggestion
/app/noelware/nino/bot/bin/bot
```
JVM args get picked up by the script via the JAVA_OPTS env var

Comment (DRSchlaubi) [reply docker/run.sh:44]: I forgot about that, but then we should export JVM_ARGS as JVM_OPTS?

Comment (NyCodeGHG) [reply docker/run.sh:44]: i think the user should just set JAVA_OPTS in their compose/helm chart/whatever

Comment (auguwu) [reply docker/run.sh:44]: > I forgot about that, but then we should export JVM_ARGS as JVM_OPTS?

Yea, I think that would be a good idea

**Review by auguwu — APPROVED** [review]
Overall, it looks fantastic, thanks Schlaubi!

**Comment by NyCodeGHG** [comment]
If i understand the gradle docs and the generated start script correctly, it has to be `JAVA_OPTS` and not `JVM_OPTS`. 
https://docs.gradle.org/current/userguide/application_plugin.html#sec:customizing_start_script_generation

**Comment by DRSchlaubi** [comment]
I shouldn't do this that late

**Pull request merged**